### PR TITLE
IP-4249-create-sso-endpoints

### PR DESCRIPTION
**Pull request by fradinni**

### CODE REVIEW

File: src/sdk/model/company/__json__/user-invitation-status-type.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export enum UserInvitationStatusEnum {
+  SENT= 'SENT',
```
Comment (iamtash) [inline src/sdk/model/company/__json__/user-invitation-status-type.ts:2]: add a space before the `=`

File: src/sdk/model/company/company.ts [hunk 2]
```diff
@@ -276,6 +288,22 @@ export class Company extends Entity {
     });
   }
 
+  /* istanbul ignore next: autogenerated */
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:291]: move 291 to 297

File: src/sdk/model/company/company.ts [hunk 3]
```diff
@@ -276,6 +288,22 @@ export class Company extends Entity {
...
+  /**
+   * Create multiple users for current company.
+   *
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:294]: remove empty line

File: src/sdk/model/company/company.ts [hunk 4]
```diff
@@ -846,4 +874,172 @@ export class Company extends Entity {
         return new CompanyLocationBackupStatus(json);
       });
   }
+
+  /**
+   * Get the identity provider for the current company.
+   * @return {Promise<IdentityProvider>} provider The identity provider
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:880]: remove the first 'provider'. sort of looks like a param reference, not needed.

File: src/sdk/model/company/company.ts [hunk 5]
```diff
@@ -846,4 +874,172 @@ export class Company extends Entity {
...
+  async getIdentityProvider(): Promise<IdentityProvider> {
+    return Iland.getHttp().get(`/companies/${this.uuid}/idp`).then((response) => {
+      const json = response.data as IdentityProviderJson;
+      return new IdentityProvider(json);
+    });
+  }
+
+  /**
+   * Get the Identity Provider redirect URI for the current company.
+   */
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:892]: add returns doc

File: src/sdk/model/company/company.ts [hunk 6]
```diff
@@ -846,4 +874,172 @@ export class Company extends Entity {
...
+  /* istanbul ignore next: autogenerated */
+  async deleteIdentityProvider(): Promise<unknown> {
+    return Iland.getHttp().delete(`/companies/${this.uuid}/idp`);
+  }
+
+  /**
+   * Create a new SSO user for current company.
+   * @param {SsoUserCreateRequest} createRequest
+   * @returns {Promise<CompanyUser>} promise Promise that resolves with the created SSO user.
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:1000]: delete the first 'promise'

File: src/sdk/model/company/company.ts [hunk 7]
```diff
@@ -846,4 +874,172 @@ export class Company extends Entity {
...
+    return Iland.getHttp().post(`/companies/${this.uuid}/sso-user`, createRequest.json).then((response) => {
+      const json = response.data as CompanyUserJson;
+      return new CompanyUser(json);
+    });
+  }
+
+  /**
+   * Create multiple SSO users for current company.
+   * @param {Array<SsoUserCreateRequest>} ssoUserCreateRequestList List of SsoUserCreateRequest.
+   * @returns {Promise<Array<CompanyUser>>} promise Promise that resolves with the list of created SSO users.
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:1013]: delete the first 'promise'

File: src/sdk/model/company/company.ts [hunk 8]
```diff
@@ -846,4 +874,172 @@ export class Company extends Entity {
...
+      .then((response) => {
+        const json = (response.data?.users || []) as Array<UserCreationJson>;
+        return json.map(it => new UserCreation(it));
+      });
+  }
+
+  /**
+   * Send a user invitation email to the given user.
+   * @param {string} username User to send invite email to
+   */
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:1029]: add returns doc

File: src/sdk/model/company/company.ts [hunk 9]
```diff
@@ -846,4 +874,172 @@ export class Company extends Entity {
...
+  /* istanbul ignore next: autogenerated */
+  async sendUserInvitationEmail(username: string): Promise<unknown> {
+    return Iland.getHttp().post(`/companies/${this.uuid}/actions/${username}/send-invite-email`);
+  }
+
+  /**
+   * Send user invitation emails to the given list of users.
+   * @param {Array<string>} userNames Users to send invite emails to
+   */
```
Comment (iamtash) [inline src/sdk/model/company/company.ts:1038]: add returns doc

File: src/sdk/model/company/user-create-request.ts [hunk 10]
```diff
@@ -36,7 +39,7 @@ export class UserCreateRequest {
    * Get password.
    * @returns {string}
```
Comment (iamtash) [inline src/sdk/model/company/user-create-request.ts:40]: | undefined

File: src/sdk/model/sso/google-identity-provider-config-request.ts [hunk 11]
```diff
@@ -0,0 +1,67 @@
+/* istanbul ignore next: autogenerated */
```
Comment (iamtash) [inline src/sdk/model/sso/google-identity-provider-config-request.ts:1]: move to right above class declaration

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+   * Get authorization url.
+   * @returns {string}
+   */
+  get authorizationUrl(): string {
+    return this._json.authorization_url;
+  }
+
+  /**
+   * Get logout URL.
+   * @returns {string}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:48]: | undefined

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+   */
+  get logoutUrl(): string | undefined {
+    return this._json.logout_url;
+  }
+
+  /**
+   * Get login hint.
+   * @returns {boolean}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:56]: | undefined

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+   */
+  get loginHint(): boolean | undefined {
+    return this._json.login_hint;
+  }
+
+  /**
+   * Get ui locales.
+   * @returns {boolean}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:64]: | undefined

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+   * Get token url.
+   * @returns {string}
+   */
+  get tokenUrl(): string {
+    return this._json.token_url;
+  }
+
+  /**
+   * Get user info url.
+   * @returns {string}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:80]: | undefined

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+   */
+  get userInfoUrl(): string | undefined {
+    return this._json.user_info_url;
+  }
+
+  /**
+   * Get backchannel supported.
+   * @returns {boolean}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:88]: | undefined

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 17]
```diff
@@ -0,0 +1,229 @@
...
+   */
+  get backchannelSupported(): boolean | undefined {
+    return this._json.backchannel_supported;
+  }
+
+  /**
+   * Get disable user info.
+   * @returns {boolean}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:96]: | undefined

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 18]
```diff
@@ -0,0 +1,229 @@
...
+   * Get client auth.
+   * @returns {OIDCClientAuthType}
+   */
+  get clientAuth(): OIDCClientAuthType {
+    return this._json.client_auth;
+  }
+
+  /**
+   * Get issuer.
+   * @returns {string}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:128]: | undefined

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+   */
+  get issuer(): string | undefined {
+    return this._json.issuer;
+  }
+
+  /**
+   * Get default scope.
+   * @returns {string}
```
Comment (iamtash) [inline src/sdk/model/sso/oidc-identity-provider-config-request.ts:136]: | undefined reflected in return doc for the rest of the getters in this class that can have return type `undefined`

File: src/sdk/model/sso/saml-identity-provider-config-request.ts [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+   * Get name id policy format.
+   * @returns {SAMLNameIdPolicyFormat}
+   */
+  get nameIsPolicyFormat(): SAMLNameIdPolicyFormat | undefined {
+    return this._json.name_id_policy_format;
+  }
+
+  /**
+   * Get principal type.
+   * @returns {SAMLPrincipalType}
```
Comment (iamtash) [inline src/sdk/model/sso/saml-identity-provider-config-request.ts:67]: | undefined  return doc for all getters in this class that have `undefined` possible return type

**Review by iamtash — CHANGES_REQUESTED** [review]

File: src/sdk/model/company/__json__/user-invitation-status-type.ts (hunk 1 above)
Comment (fradinni) [reply src/sdk/model/company/__json__/user-invitation-status-type.ts]: Fixed

File: src/sdk/model/company/company.ts (hunk 2 above)
Comment (fradinni) [reply src/sdk/model/company/company.ts]: Fixed

File: src/sdk/model/company/company.ts (hunk 4 above)
Comment (fradinni) [reply src/sdk/model/company/company.ts]: Fixed

File: src/sdk/model/company/company.ts (hunk 5 above)
Comment (fradinni) [reply src/sdk/model/company/company.ts:892]: Fixed

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts (hunk 16 above)
Comment (fradinni) [reply src/sdk/model/sso/oidc-identity-provider-config-request.ts]: Fixed

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts (hunk 17 above)
Comment (fradinni) [reply src/sdk/model/sso/oidc-identity-provider-config-request.ts]: Fixed

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts (hunk 18 above)
Comment (fradinni) [reply src/sdk/model/sso/oidc-identity-provider-config-request.ts]: Fixed

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts (hunk 19 above)
Comment (fradinni) [reply src/sdk/model/sso/oidc-identity-provider-config-request.ts]: Fixed

File: src/sdk/model/sso/saml-identity-provider-config-request.ts (hunk 20 above)
Comment (fradinni) [reply src/sdk/model/sso/saml-identity-provider-config-request.ts]: Fixed

**Comment by iamtash** [comment]
hey @fradinni still have a few minor unresolved comments here

File: src/sdk/model/company/company.ts (hunk 6 above)
Comment (fradinni) [reply src/sdk/model/company/company.ts:1000]: Done

File: src/sdk/model/company/company.ts (hunk 7 above)
Comment (fradinni) [reply src/sdk/model/company/company.ts:1013]: Done

File: src/sdk/model/company/user-create-request.ts (hunk 10 above)
Comment (fradinni) [reply src/sdk/model/company/user-create-request.ts:40]: Done

File: src/sdk/model/sso/google-identity-provider-config-request.ts (hunk 11 above)
Comment (fradinni) [reply src/sdk/model/sso/google-identity-provider-config-request.ts:1]: Done

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts (hunk 12 above)
Comment (fradinni) [reply src/sdk/model/sso/oidc-identity-provider-config-request.ts]: Done

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts (hunk 13 above)
Comment (fradinni) [reply src/sdk/model/sso/oidc-identity-provider-config-request.ts]: Done

File: src/sdk/model/sso/oidc-identity-provider-config-request.ts (hunk 15 above)
Comment (fradinni) [reply src/sdk/model/sso/oidc-identity-provider-config-request.ts]: Done

**Comment by fradinni** [comment]
Thank you @iamtash ! Should be good now.

**Review by iamtash — APPROVED** [review]

**Pull request merged**